src/api/scripts/scripts.api.ts

```diff
@@ -98,12 +99,31 @@ export async function fetchScriptByNameAndVersionDownload({
  * Makes a new 'script-version-combo', which can either be the first version of a totally new script,
  * OR an extra version of an existing script.
  */
-export function createScriptVersion(script: IScriptBase) {
+export function createScriptVersion(script: IScriptBase | IScriptImport) {
+    if (!instanceOfIScriptImport(script)) {
+        return post<IScriptBase>({
+            needsAuthentication: true,
+            isIesiApi: true,
+            url: API_URLS.SCRIPTS,
+            body: script,
+        });
+    }
     return post<IScriptBase>({
         needsAuthentication: true,
         isIesiApi: true,
         url: API_URLS.SCRIPTS,
-        body: script,
+        body: script.value,
+        contentType: script.value instanceof FormData ? 'multipart/form-data' : 'application/json',
+        headers: {
+            'Content-Type': script.value instanceof FormData ? 'multipart/form-data' : 'application/json',
+        },
```

Comment:
Before checking if it is a FormData type, check also whether it is an import or not